wordpress-mobile/WordPress-Android
# Add reblog to reader post list

Fixes #11532 

## Description

This feature adds reblog action to post cards in stream view

A reblog button has been added in the actions of the post cards in stream view.
If the user has more that one sites the button triggers the existing Site Picker.

| Post Details |Site Picker (unchanged)|
|--- |---|
|![device-2020-03-27-082924](https://user-images.githubusercontent.com/304044/77728549-6b907c80-7005-11ea-925e-d361de9e894d.png)|![2020-03-19 22 05 33](https://user-images.githubusercontent.com/304044/77110683-9b041f80-6a2e-11ea-8f21-198d89d5ead1.png)|

If the user has only one site the editor is triggered directly.
At the moment handling users with no site is not supported.

When the users selects a site he is redirected to the editor with the following information prefilled:
* Title (with source)
* Image (if the post has a featured)
* Quote
* Citation

| Gutenberg editor with featured image| Gutenberg editor without image |
| --- | --- |
|![device-2020-03-22-114527](https://user-images.githubusercontent.com/304044/77246840-b8560b00-6c33-11ea-8b38-28dc15e7b888.png)|![2020-03-19 22 06 05](https://user-images.githubusercontent.com/304044/77110686-9c354c80-6a2e-11ea-858d-f09c597f6348.png)|

| Aztec editor with featured image| Aztec editor without image |
| --- | --- |
|![device-2020-03-22-114630](https://user-images.githubusercontent.com/304044/77246853-d459ac80-6c33-11ea-9059-40c0c62eb03f.png)|![device-2020-03-22-114609](https://user-images.githubusercontent.com/304044/77246855-d885ca00-6c33-11ea-8bea-9943adfebb79.png)|

When the user taps the Publish button and confirms he returns to the original context

| Uploading State| Post Published |
| --- | --- |
|![device-2020-03-27-083017](https://user-images.githubusercontent.com/304044/77728611-8531c400-7005-11ea-8091-14d377717765.png)|![device-2020-03-27-083046](https://user-images.githubusercontent.com/304044/77728603-82cf6a00-7005-11ea-8f5b-04b9d3914967.png)|

## Test

0. Open the App and Login if needed
1. Select the Reader tab
2. Notice the reblog button on the post cards actions
3. Tap on the reblog action on one of the post cards
4. If prompted select a site from the picker
5. Notice that the editor opens with content from the original post: Title (with source), Image (if the post has a featured), Quote and Citation link
6. Optionally edit the post
7. Press the Publish button
8. Confirm by pressing the Publish Now button in the dialog
9. Notice that you return to the original context (post list view)
10. Notice the snackbar with the VIEW button
11. Tap on the VIEW button
12. Examine the reblogged post in the browser

PR submission checklist:
